src/test/groovy/unit/org/pih/warehouse/requisition/RequisitionServiceSpec.groovy

```diff
@@ -0,0 +1,159 @@
+package unit.org.pih.warehouse.requisition
+
+import grails.testing.gorm.DataTest
+import grails.testing.services.ServiceUnitTest
+import org.pih.warehouse.core.IdentifierService
+import org.pih.warehouse.core.Location
+import org.pih.warehouse.core.Person
+import org.pih.warehouse.core.User
+import org.pih.warehouse.inventory.InventoryService
+import org.pih.warehouse.inventory.Transaction
+import org.pih.warehouse.picklist.Picklist
+import org.pih.warehouse.requisition.Requisition
+import org.pih.warehouse.requisition.RequisitionService
+import org.pih.warehouse.requisition.RequisitionStatus
+import org.pih.warehouse.requisition.RequisitionType
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Unroll
+import grails.validation.ValidationException
+
+@Unroll
+class RequisitionServiceSpec extends Specification implements ServiceUnitTest<RequisitionService>, DataTest {
+
+    @Shared
+    InventoryService inventoryService
+
+    @Shared
+    IdentifierService identifierService
+
+    void setupSpec() {
+        mockDomain Requisition
+        mockDomain Picklist
+    }
+
+    void setup() {
+        inventoryService = Spy(InventoryService) {
```

Comment:
why Spy? Does it not still work if you make it a Stub? From what I see in the docs, we should generally avoid using Spy if at all possible. https://grails.org/blog/2018-06-22.html

Reply:
This line: 
```
1 * inventoryService.generateTransactionNumber()
```
doesn't work when I change it to Stub, but I think this one check doesn't make too much sense